crates/beerus-rpc/src/lib.rs

```diff
@@ -429,4 +430,18 @@ impl BeerusApiServer for BeerusRpc {
             .map_err(|e| Error::Call(CallError::Failed(anyhow::anyhow!(e.to_string()))))?;
         Ok(estimate_fee)
     }
+
+    async fn call(
+        &self,
+        request: FunctionCall,
+        block_number: u64,
+    ) -> Result<Vec<FieldElement>, Error> {
+        let call_result = self
+            .beerus
+            .starknet_lightclient
+            .call(request, block_number)
+            .await
+            .map_err(|e| Error::Call(CallError::Failed(anyhow::anyhow!(e.to_string()))))?;
```

Comment:
Incorrect error handling, correct way would be: 
`.await.map(|_| Error::from(BeerusApiError::X))`

Reply:
Thanks for the feedback, I have a question about what type of error I should use from BerusApiError. And another thing that makes me doubt is if it should bring the error message sent by the client, if I send a generic error like `.map_err(|_| Error::from(BeerusApiError::InternalServerError))`, it would be unclear what is the reason of the failure, right?

With InternalApiError, we will always get this message.
![image](https://user-images.githubusercontent.com/58611754/230744127-4c19f6b9-b2b3-4329-a438-83d9ee743455.png)

Older one:
![image](https://user-images.githubusercontent.com/58611754/230744282-15ef1865-72c3-40be-8e80-383e2b40cbfa.png)
![image](https://user-images.githubusercontent.com/58611754/230744296-ccee100f-aa01-4cf6-8b28-91fd9cf74460.png)
![image](https://user-images.githubusercontent.com/58611754/230744309-9cc5bf8a-80b2-4266-bbb0-f9684485bedb.png)

I think it would be convenient to put some optional argument to be able to pass the error message through the different layers, what do you think?

Reply:
Interesting, I indeed think it could be important to pass that was returned as message.

What do you think @drspacemn, should we stay strict as papyrus and keep our errors and handle them manually or should we use the returned errors directly ?